### ISSUE TITLE
ApplicationInsights DependencyCollector warning

### ISSUE THREAD

**Issue by rfcdejong** [Xabaril/AspNetCore.Diagnostics.HealthChecks#86]
Including package
<PackageReference Include="AspNetCore.HealthChecks.Publisher.ApplicationInsights" Version="2.2.1" />

Results into this warning

warning NU1608: Detected package version outside of dependency constraint: Microsoft.ApplicationInsights.DependencyCollector 2.4.1 requires Microsoft.ApplicationInsights (= 2.4.0) but version Microsoft.ApplicationInsights 2.7.2 was resolved.

Please set a dependency to ApplicationInsights >= 2.4.0 and do not force a need for assemblybinding configuration.

PS: This only happens on a common library project 'Microsoft.NET.Sdk' and not on the WebAPI projects somehow.. 'Microsoft.NET.Sdk.Web'

**Comment by unaizorrilla** [comment]
Can your share a project with this warning to check versions and dependencies?

**Comment by rfcdejong** [comment]
Sorry, but the project cannot be shared as it is not closed source, proprietary and confidential.. but I can list the PackageReferences, I removed the faulting package causing the warning cause we do not Publish it currently.

`    <PackageReference Include="AspNetCore.HealthChecks.Publisher.ApplicationInsights" Version="2.2.1" />
`
```
    <PackageReference Include="AspNetCore.HealthChecks.AzureStorage" Version="2.2.1" />
    <PackageReference Include="AspNetCore.HealthChecks.OpenIdConnectServer" Version="2.2.2" />
    <PackageReference Include="AspNetCore.HealthChecks.Redis" Version="2.2.2" />
    <PackageReference Include="Microsoft.AspNetCore.All" />
    <PackageReference Include="Microsoft.Diagnostics.EventFlow.Inputs.Serilog" Version="1.4.0" />
    <PackageReference Include="Microsoft.Diagnostics.EventFlow.ServiceFabric" Version="1.5.0" />
    <PackageReference Include="Microsoft.ServiceFabric" Version="6.1.480" />
    <PackageReference Include="Microsoft.ServiceFabric.Actors" Version="3.0.480" />
    <PackageReference Include="Microsoft.ServiceFabric.AspNetCore.Abstractions" Version="3.0.480" />
    <PackageReference Include="Microsoft.ServiceFabric.AspNetCore.HttpSys" Version="3.0.480" />
    <PackageReference Include="Microsoft.ServiceFabric.AspNetCore.Kestrel" Version="3.0.480" />
    <PackageReference Include="Microsoft.ServiceFabric.Data" Version="3.0.480" />
    <PackageReference Include="Microsoft.ServiceFabric.Diagnostics.Internal" Version="3.0.480" />
    <PackageReference Include="Microsoft.ServiceFabric.FabricTransport.Internal" Version="3.0.480" />
    <PackageReference Include="Microsoft.ServiceFabric.Services" Version="3.0.480" />
    <PackageReference Include="Microsoft.ServiceFabric.Services.Remoting" Version="3.0.480" />
    <PackageReference Include="Newtonsoft.Json" Version="11.0.2" />
    <PackageReference Include="RestSharp" Version="106.5.4" />
    <PackageReference Include="Serilog" Version="2.8.0" />
    <PackageReference Include="Serilog.Extensions.Logging" Version="2.0.2" />
    <PackageReference Include="Serilog.Sinks.Console" Version="3.1.1" />
    <PackageReference Include="Serilog.Sinks.Literate" Version="3.0.0" />
    <PackageReference Include="Serilog.Sinks.Observable" Version="2.0.2" />    
    <PackageReference Include="Swashbuckle.AspNetCore" Version="4.0.1" />
    <PackageReference Include="Swashbuckle.AspNetCore.Swagger" Version="4.0.1" />
    <PackageReference Include="Swashbuckle.AspNetCore.SwaggerGen" Version="4.0.1" />
    <PackageReference Include="Swashbuckle.AspNetCore.SwaggerUI" Version="4.0.1" />
    <PackageReference Include="System.Reflection.Extensions" Version="4.3.0" />
```
